homeassistant/components/recorder/migration.py

```diff
@@ -41,20 +41,39 @@ def _apply_update(engine, new_version):
     from sqlalchemy import Table
     from . import models
 
-    if new_version == 1:
-        def create_index(table_name, column_name):
-            """Create an index for the specified table and column."""
-            table = Table(table_name, models.Base.metadata)
-            name = "_".join(("ix", table_name, column_name))
-            # Look up the index object that was created from the models
-            index = next(idx for idx in table.indexes if idx.name == name)
-            _LOGGER.debug("Creating index for table %s column %s",
-                          table_name, column_name)
-            index.create(engine)
-            _LOGGER.debug("Index creation done for table %s column %s",
-                          table_name, column_name)
+    def create_index(table_name, column_name):
+        """Create an index for the specified table and column."""
+        table = Table(table_name, models.Base.metadata)
+        name = "_".join(("ix", table_name, column_name))
+        # Look up the index object that was created from the models
+        index = next(idx for idx in table.indexes if idx.name == name)
+        _LOGGER.debug("Creating index for table %s column %s",
+                      table_name, column_name)
+        index.create(engine)
+        _LOGGER.debug("Index creation done for table %s column %s",
+                      table_name, column_name)
+
+    def create_compound_index(table_name, column_names):
+        """Create an index for the specified table and columns."""
+        table = Table(table_name, models.Base.metadata)
+        index_name = "ix_" + table_name + "_" + "_".join(column_names)
```

Comment:
Hmm in terms of the index naming, would it make more sense to use literals here? If we have to manually create the index name on models.py, it seems like it might be easier to do that here? So something like `create_index('recorder_runs', 'ix_recorder_runs_start_end')` given what is in models.py is `    __table_args__ = (Index('ix_recorder_runs_start_end', 'start', 'end'),)`

I'm normally not a fan of literals like that, but unless there is a way to auto-generate an index name in models.py, it might keep things a bit less error-prone for folks that don't know about the auto-naming bits going on in migration.py?